pkg/kubectl/cmd/util/factory.go

```diff
@@ -301,9 +323,36 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return priorityRESTMapper, api.Scheme
 		},
+		DynamicObject: func() (meta.RESTMapper, runtime.ObjectTyper) {
```

Comment:
>This should probably return an error (that a command has to handle) - nothing else in factory calls CheckErr(). Does it complicate where this will be called from?

I was actually only following the pattern in other factory calls. `factory.Object` has 6 calls to `CheckErr`. I agree that both of these should return errors instead, but I think that's a separate issue.

>I do think we want to make both of these interfaces somewhat lazy (only load config when asked, and cache). If we do it this way, will we always be doing server discovery in every get call? Does that make normal get significantly worse in terms of number of requests it is making to the server?

So far for the commands I've converted, there's only one call to `DynamicObject`. We could cache it, but I don't yet see any command that would benefit from it. With this change, every converted kubectl command will do discovery just to build this object.